miner/worker.go

```diff
@@ -1293,6 +1326,33 @@ func (w *worker) commitWork(ctx context.Context, interrupt *int32, noempty bool,
 	w.current = work
 }
 
+func getInterruptTimer(ctx context.Context, work *environment, current *types.Block) (chan struct{}, func()) {
+	delay := time.Until(time.Unix(int64(work.header.Time), 0))
+
+	timeoutTimer := time.NewTimer(delay)
+	stopFn := func() {
+		timeoutTimer.Stop()
+	}
+
+	blockNumber := current.NumberU64() + 1
+	interruptCh := make(chan struct{})
+
+	go func() {
+		select {
+		case <-timeoutTimer.C:
+			log.Info("an interrupt event. cancelling the current block",
+				"block", blockNumber,
+				"hash", current.Hash().Hex(),
```

Comment:
Do you expect to have block number from the next block, but block number from the previous one?